Loading and Preprocessing:

In [1]:
import pandas as pd
import mlxtend
from mlxtend.frequent_patterns import apriori, association_rules

In [ ]:
df=pd.read_excel('/content/AnalysisData (1).xlsx')
df.head()

Transaction Id Product 1 name Product 2 name Product 3 name
0               1          Apple         Potato         Banana
1               2          Apple         Potato      Ice cream
2               3          Bread           Milk         Butter
3               4          Apple           Milk        Almonds
4               5          Apple         Onions         Potato

One-Hot Encoding:


In [ ]:
df1=pd.get_dummies(df)
df2=df1.iloc[:,1:]
df2

Product 1 name_Apple  Product 1 name_Banana  ...  Product 3 name_Tea  Product 3 name_Tie
0                      1                      0  ...                   0                   0
1                      1                      0  ...                   0                   0
2                      0                      0  ...                   0                   0
3                      1                      0  ...                   0                   0
4                      1                      0  ...                   0                   0
5                      0                      0  ...                   0                   1
6                      0                      1  ...                   0                   0
7                      0                      1  ...                   0                   0
8                      0                      0  ...                   0                   0
9                      0                      0  ...                   0                   0
10                     0                      0  ...                   0                   0
11                     0                      0  ...                   1                   0
12                     0                      0  ...                   0                   0
13                     0                      0  ...                   0                   0

[14 rows x 31 columns]

Apriori Algorithm:

In [ ]:
frequentitems=apriori(df2,min_support=0.07,use_colnames=True)
frequentitems

support                                           itemsets
0   0.285714                             (Product 1 name_Apple)
1   0.142857                            (Product 1 name_Banana)
2   0.142857                             (Product 1 name_Bread)
3   0.142857                              (Product 1 name_Milk)
4   0.071429                               (Product 1 name_Oil)
..       ...                                                ...
81  0.071429  (Product 2 name_Tea, Product 3 name_Bread, Pro...
82  0.071429  (Product 2 name_Potato, Product 1 name_Oil, Pr...
83  0.071429  (Product 3 name_Oil, Product 2 name_Garlic, Pr...
84  0.071429  (Product 3 name_Tie, Product 2 name_Tshirts, P...
85  0.071429  (Product 2 name_Tie, Product 1 name_Tshirts, P...

[86 rows x 2 columns]

Apriori Algorithm:

In [ ]:
rule=association_rules(frequentitems,metric="lift",min_threshold=1)
rule

antecedents  ... zhangs_metric
0                            (Product 1 name_Apple)  ...      0.600000
1                             (Product 2 name_Milk)  ...      0.500000
2                            (Product 1 name_Apple)  ...      1.000000
3                           (Product 2 name_Onions)  ...      0.769231
4                            (Product 1 name_Apple)  ...      0.800000
..                                              ...  ...           ...
161      (Product 2 name_Tie, Product 3 name_Shoes)  ...      1.000000
162  (Product 3 name_Shoes, Product 1 name_Tshirts)  ...      1.000000
163                            (Product 2 name_Tie)  ...      1.000000
164                        (Product 1 name_Tshirts)  ...      1.000000
165                          (Product 3 name_Shoes)  ...      1.000000

[166 rows x 10 columns]

Filtering Rules by Lift:

In [ ]:
rule[(rule["lift"]==3) ]
rule

antecedents  ... zhangs_metric
0                            (Product 1 name_Apple)  ...      0.600000
1                             (Product 2 name_Milk)  ...      0.500000
2                            (Product 1 name_Apple)  ...      1.000000
3                           (Product 2 name_Onions)  ...      0.769231
4                            (Product 1 name_Apple)  ...      0.800000
..                                              ...  ...           ...
161      (Product 2 name_Tie, Product 3 name_Shoes)  ...      1.000000
162  (Product 3 name_Shoes, Product 1 name_Tshirts)  ...      1.000000
163                            (Product 2 name_Tie)  ...      1.000000
164                        (Product 1 name_Tshirts)  ...      1.000000
165                          (Product 3 name_Shoes)  ...      1.000000

[166 rows x 10 columns]

Rule 0:

Antecedents: Product 1 name_Apple
Consequents: Product 2 name_Milk
Support: 0.071429
Confidence: 0.25
Lift: 1.75
Interpretation: There is a 25% chance of buying Milk when Apple is purchased, and this association has a lift of 1.75, indicating a positive correlation between buying Apple and Milk.

Rule 2:

Antecedents: Product 1 name_Apple
Consequents: Product 2 name_Onions
Support: 0.071429
Confidence: 0.25
Lift: 3.5
Interpretation: There is a 25% chance of buying Onions when Apple is purchased, and the high lift of 3.5 suggests a strong positive association between buying Apple and Onions.
